### ISSUE TITLE
Nesting freeze_time doesn't work with time.time()

### ISSUE THREAD

**Issue by joegilley** [spulec/freezegun#113]
```
@freeze_time('2015-01-01')
def test_method(self):
    print time(), datetime.utcnow()
    with freeze_time('2015-01-01T00:06:00'):
        print time(), datetime.utcnow()
```

Yields:

```
1420070400.0 2015-01-01 00:00:00
1420070400.0 2015-01-01 00:06:00
```

```
$ pip show freezegun

---
Name: freezegun
Version: 0.3.5
Location: /home/____/paxon/env/lib/python2.7/site-packages
Requires: six, python-dateutil
```

**Comment by joegilley** [comment]
bump.

**Comment by spulec** [comment]
I have confirmed this is a bug. Due to how time is mocked (compared to datetime), the fix is going to take a bit of work.

**Comment by yelled3** [comment]
@spulec IMO, this is a fairly important issue - is there any progress with this?

**Comment by spulec** [comment]
No progress, feel free to take a stab at it.

**Comment by neilsh** [comment]
Generally speaking, what's the difference between how the two are mocked, and what would need to change?

**Comment by pablodo** [comment]
Im experiencing something similar. When freezing time, `time.time` returns the timestamp in `UTC`, and `datetime` returns it with my local timezone:

```
>>> with freezegun.freeze_time('2016-11-21 15:00:00'):
...     print datetime.now()
...     print datetime.fromtimestamp(time.time())
... 
2016-11-21 15:00:00
2016-11-21 12:00:00
```

**Comment by 3h4x** [comment]
+1

**Comment by azmeuk** [comment]
Also, when a tz_offset is provided, it is ignored.
```python
>>> with freezegun.freeze_time('2016-11-21 15:00:00', tz_offset=6):
...     print (datetime.now())
...     print (datetime.fromtimestamp(time.time()))
... 
2016-11-21 21:00:00
2016-11-21 12:00:00
```

**Comment by boxed** [comment]
From a very cursory inspection of api.py it looks like this is the start of the problem:

```python
        datetime.datetime = FakeDatetime
        datetime.date = FakeDate
        fake_time = FakeTime(time_to_freeze, time.time)
        fake_localtime = FakeLocalTime(time_to_freeze, time.localtime)
        fake_gmtime = FakeGMTTime(time_to_freeze, time.gmtime)
        fake_strftime = FakeStrfTime(time_to_freeze, time.strftime)
```

and later

```python
        datetime.datetime.times_to_freeze.append(time_to_freeze)
        datetime.datetime.tz_offsets.append(self.tz_offset)

        datetime.date.dates_to_freeze.append(time_to_freeze)
        datetime.date.tz_offsets.append(self.tz_offset)
```

The nesting works because the second block, which is not implemented for the birst blocks. It looks like the code could be simplified a lot if this was unified too, so instead of 

```
        datetime.date.dates_to_freeze.append(time_to_freeze)
        datetime.date.tz_offsets.append(self.tz_offset)
```

you could have

```
for foo in fakes:
    foo.times_to_freeze.append(time_to_freeze)
    foo.tz_offsets.append(self.tz_offset)
```

or similar but better named obviously.

**Comment by spulec** [comment]
Closed with #260
